go/client/cmd_ctl_watchdog_windows.go

```diff
@@ -50,44 +50,54 @@ func (c *CmdWatchdog) Run() (err error) {
 	//
 	// Note that we give up after 10 consecutive crashes
 
-	if c.G().SocketInfo == nil {
-		return errors.New("Uninitialized socket")
-	}
-	pipeName := c.G().SocketInfo.GetFile()
-	if len(pipeName) == 0 {
-		return errors.New("No pipe name")
-	}
-
 	countdown := c.restarts
 	for {
-		conn, err := npipe.DialTimeout(pipeName, time.Second*10)
-		if conn == nil {
-			// no service started. exit.
+		// Blocking wait on service. First, there has to be a pid
+		// file, because this is a forking command.
+		var fn string
+		if fn, err = c.G().Env.GetPidFile(); err != nil {
```

Comment:
Yes the way it is written, it forkes it from the get-go, since it is
implemented as an auto-fork command. But there could be cases where it is
already running, which should be OK too.

Now I'm getting confused what we should do vs. what we're already doing vs.
what we could do.

On Wed, Nov 18, 2015 at 10:21 AM, Maxwell Krohn <notifications@github.com>
wrote:

> In go/client/cmd_ctl_watchdog_windows.go
> <https://github.com/keybase/client/pull/1349#discussion_r45237931>:
>
> >  	countdown := c.restarts
> >  	for {
> > -		conn, err := npipe.DialTimeout(pipeName, time.Second*10)
> > -		if conn == nil {
> > -			// no service started. exit.
> > +		// Blocking wait on service. First, there has to be a pid
> > +		// file, because this is a forking command.
> > +		var fn string
> > +		if fn, err = c.G().Env.GetPidFile(); err != nil {
>
> so maybe this is what you were saying before, but maybe the watchdog
> should fork the service from the get-go? This seems cleaner since then all
> cases are the same, not having the first case be an exception.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1349/files#r45237931>.
>